src/aws_encryption_sdk/keyring/raw_keyring.py

```diff
@@ -0,0 +1,390 @@
+            )
+        else:
+            # Encrypt data key
+            encrypted_wrapped_key = EncryptedData(
+                iv=None,
+                ciphertext=self._wrapping_key.public_key().encrypt(
+                    plaintext=plaintext_data_key, padding=self._wrapping_algorithm.padding
+                ),
+                tag=None,
+            )
+
+        # EncryptedData to EncryptedDataKey
+        encrypted_data_key = serialize_wrapped_key(
+            key_provider=self._key_provider,
+            wrapping_algorithm=self._wrapping_algorithm,
+            wrapping_key_id=self.key_name,
+            encrypted_wrapped_key=encrypted_wrapped_key,
+        )
+
+        # Update Keyring Trace
+        if encrypted_data_key:
+            keyring_trace = KeyringTrace(
+                wrapping_key=encrypted_data_key.key_provider, flags={KeyringTraceFlag.WRAPPING_KEY_ENCRYPTED_DATA_KEY}
+            )
+
+            # Add encrypted data key to encryption_materials
+            encryption_materials.add_encrypted_data_key(
+                encrypted_data_key=encrypted_data_key, keyring_trace=keyring_trace
+            )
+
+        return encryption_materials
+
+    def on_decrypt(self, decryption_materials, encrypted_data_keys):
+        # type: (DecryptionMaterials, Iterable[EncryptedDataKey]) -> DecryptionMaterials
+        """Attempt to decrypt the encrypted data keys.
+
+        :param decryption_materials: Decryption materials for the keyring to modify.
+        :type decryption_materials: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        :param encrypted_data_keys: List of encrypted data keys.
+        :type: List of `aws_encryption_sdk.structures.EncryptedDataKey`
+        :returns: Optionally modified decryption materials.
+        :rtype: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        """
+        if isinstance(self._wrapping_key, rsa.RSAPrivateKey):
+            # Decrypt data key
+            for key in encrypted_data_keys:
+                if key.key_provider == self._key_provider:
+                    # Wrapped EncryptedDataKey to deserialized EncryptedData
+                    encrypted_wrapped_key = deserialize_wrapped_key(
+                        wrapping_algorithm=self._wrapping_algorithm,
+                        wrapping_key_id=self.key_name,
+                        wrapped_encrypted_key=key,
+                    )
+                    try:
+                        plaintext_data_key = self._wrapping_key.decrypt(
+                            ciphertext=encrypted_wrapped_key.ciphertext, padding=self._wrapping_algorithm.padding
+                        )
+                    except Exception as error:  # pylint: disable=broad-except
+                        logger = logging.getLogger()
+                        logger.error(error.__class__.__name__, ":", str(error))
+                        return decryption_materials
+
+                    if plaintext_data_key:
```

Comment:
What is the logic flow where we have arrived at this line but this check is false?